app/ts/background/medataSearch.ts

```diff
@@ -73,14 +73,20 @@ const convertContractDefinitionToAddressBookEntry = ([address, def]: [string, Co
 	entrySource: 'DarkFloristMetadata' as const,
 })
 
+function concatArraysUniqueByAddress<T>(addTo: readonly (T & { address: bigint })[], addFrom: readonly (T & { address: bigint })[]) {
```

Comment:
not sure if `T` here should also be cast within the function but can alternatively use ReadonlyArray as well

```suggestion
function concatArraysUniqueByAddress<T>(addTo: ReadonlyArray<T & { address: bigint }>, addFrom: ReadonlyArray<T & { address: bigint }>) {
```

Reply:
these are the same thigns